build.js

```diff
@@ -117,345 +117,578 @@ var getModuleConfig = function(api, app) {
-  }
+      .return(moduleConfig);
+  };
+
+  var getTextBetween = function(text, startToken, endToken) {
+    var start = text.indexOf(startToken);
+    var end = text.indexOf(endToken);
+    if (start == -1 || end == -1) {
+      return "";
+    }
+    var offset = text.substring(start).indexOf("\n") + 1;
+    var afterStart = start + offset;
+    return text.substring(afterStart, end);
+  };
+
+  var replaceTextBetween = function(text, startToken, endToken, replaceText) {
+    var newText = "";
+    var start = text.indexOf(startToken);
+    var end = text.indexOf(endToken);
+    if (start == -1 || end == -1) {
+      return text;
+    }
+    var offset = text.substring(start).indexOf("\n") + 1;
+    var afterStart = start + offset;
+    newText += text.substring(0, afterStart);
+    newText += replaceText;
+    newText += text.substring(end);
+
+    return newText;
+  };
+
+  function injectAppLinks(android_manifest, app) {
+    var manifestXml = path.join(projectPath ,"/app/src/main",  'AndroidManifest.xml');
+    var app_links = android_manifest.app_links;
+    var template = '<data android:host="curr_host" android:scheme="curr_scheme"/>';
+    var result = '';
+
+    if (!app_links || app_links.length === 0) {
+      return Promise.resolve(true);
+    }
+
+    app_links.forEach(function (curr) {
+      var curr_data = '';
+      var host = curr.host;
+      var scheme = curr.scheme || 'http';
 
-  app_links.forEach(function (curr) {
-    var curr_data = '';
-    var host = curr.host;
-    var scheme = curr.scheme || 'http';
+      curr_data = template.replace('curr_host', host);
+      curr_data = curr_data.replace('curr_scheme', scheme);
+      ['path', 'pathPrefix', 'pathPattern', 'port'].forEach(function (prop) {
+        if (curr[prop]) {
+          curr_data = curr_data.replace('/>', ' android:'.concat(prop, '="', curr[prop], '"/>'))
+        }
+      });
+      result = result.concat(curr_data);
+    });
+    result = result.concat('\n');
+
+    return Promise.resolve()
+      .then(function () {
+        return fs.readFileAsync(manifestXml, 'utf-8');
+      })
+      .then(function (xml) {
+        var XML_START_PLUGINS_LINKS = '<!--START_PLUGINS_LINKS-->';
+        var XML_END_PLUGINS_LINKS = '<!--END_PLUGINS_LINKS-->';
+
+        xml = replaceTextBetween(xml, XML_START_PLUGINS_LINKS, XML_END_PLUGINS_LINKS, result);
+
+        return fs.writeFileAsync(manifestXml, xml, 'utf-8');
+      });
+  };
 
-    curr_data = template.replace('curr_host', host);
-    curr_data = curr_data.replace('curr_scheme', scheme);
-    ['path', 'pathPrefix', 'pathPattern', 'port'].forEach(function (prop) {
-      if (curr[prop]) {
-        curr_data = curr_data.replace('/>', ' android:'.concat(prop, '="', curr[prop], '"/>'))
+
+
+  function injectPluginXML(opts, app) {
+    var moduleConfig = opts.moduleConfig;
+    //var outputPath = opts.outputPath;
+    var manifestXml =  path.join(projectPath,"app/src/main",  'AndroidManifest.xml');
```

Comment:
manifest file path is obtained in above function as well, hence can be moved to seperate function, say getManifestFilePath